src/rust/engine/src/externs/options.rs

```diff
@@ -146,6 +149,50 @@ impl PyOptionId {
     }
 }
 
+#[pyclass]
+struct PySplitArgs(SplitArgs);
+
+#[pymethods]
+impl PySplitArgs {
+    fn goals(&self) -> &Vec<String> {
+        &self.0.goals
+    }
+
+    fn unknown_goals(&self) -> &Vec<String> {
+        &self.0.unknown_goals
+    }
+
+    fn specs(&self) -> &Vec<String> {
+        &self.0.specs
+    }
+
+    fn passthru(&self) -> &Vec<String> {
+        &self.0.passthru
+    }
+}
+
+#[pyclass]
+struct PyArgSplitter(ArgSplitter);
+
+#[pymethods]
+impl PyArgSplitter {
+    #[new]
+    fn __new__(build_root: &str, known_goal_names: Vec<String>) -> PyResult<Self> {
+        Ok(Self(ArgSplitter::new(
+            &PathBuf::from(build_root),
+            known_goal_names
+                .iter()
+                .map(AsRef::as_ref)
+                .collect::<Vec<_>>(),
+        )))
+    }
+
+    #[pyo3(signature = (args))]
+    fn split_args(&self, args: Vec<String>) -> PyResult<PySplitArgs> {
```

Comment:
Simpler would be to omit `PyResult` since this always returns `Ok`. That is, just return `PySplitArgs` and PyO3 will do the right thing.

Reply:
Same here.